cmd/kubectl-gadget/audit/seccomp.go

```diff
@@ -19,64 +19,83 @@ import (
+	seccompauditTypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/audit/seccomp/types"
 	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
-
-	"github.com/spf13/cobra"
 )
 
 func newSeccompCmd() *cobra.Command {
 	var commonFlags utils.CommonFlags
 
-	cmd := &cobra.Command{
-		Use:          "seccomp",
-		Short:        "Audit syscalls according to the seccomp profile",
-		Args:         cobra.NoArgs,
-		SilenceUsage: true,
-		RunE: func(cmd *cobra.Command, args []string) error {
-			parser := commonaudit.NewSeccompK8sParser(&commonFlags.OutputConfig)
-
-			if commonFlags.OutputMode != commonutils.OutputModeJSON {
-				fmt.Println(parser.BuildColumnsHeader())
+	runCmd := func(cmd *cobra.Command, args []string) error {
+		parser, err := commonutils.NewGadgetParserWithK8sInfo(
+			&commonFlags.OutputConfig,
+			seccompauditTypes.GetColumns(),
+		)
+		if err != nil {
+			return commonutils.WrapInErrParserCreate(err)
+		}
+
+		if commonFlags.OutputMode != commonutils.OutputModeJSON {
+			fmt.Println(parser.BuildColumnsHeader())
+		}
+
+		config := &utils.TraceConfig{
+			GadgetName:       "audit-seccomp",
+			Operation:        gadgetv1alpha1.OperationStart,
+			TraceOutputMode:  gadgetv1alpha1.TraceOutputModeStream,
+			TraceOutputState: gadgetv1alpha1.TraceStateStarted,
+			CommonFlags:      &commonFlags,
+		}
+
+		transformEvent := func(line string) string {
+			var e seccompauditTypes.Event
+
+			if err := json.Unmarshal([]byte(line), &e); err != nil {
+				fmt.Fprintf(os.Stderr, "Error: %s", commonutils.WrapInErrUnmarshalOutput(err, line))
+				return ""
 			}
 
-			config := &utils.TraceConfig{
-				GadgetName:       "audit-seccomp",
-				Operation:        gadgetv1alpha1.OperationStart,
-				TraceOutputMode:  gadgetv1alpha1.TraceOutputModeStream,
-				TraceOutputState: gadgetv1alpha1.TraceStateStarted,
-				CommonFlags:      &commonFlags,
+			baseEvent := e.Event
+			if baseEvent.Type != eventtypes.NORMAL {
+				commonutils.ManageSpecialEvent(baseEvent, commonFlags.Verbose)
+				return ""
 			}
 
-			transformEvent := func(line string) string {
-				var e types.Event
-
-				if err := json.Unmarshal([]byte(line), &e); err != nil {
-					fmt.Fprintf(os.Stderr, "Error: %s", commonutils.WrapInErrUnmarshalOutput(err, line))
+			switch commonFlags.OutputMode {
+			case commonutils.OutputModeJSON:
+				b, err := json.Marshal(e)
+				if err != nil {
+					fmt.Fprint(os.Stderr, fmt.Sprint(commonutils.WrapInErrMarshalOutput(err)))
 					return ""
 				}
 
-				if e.Type != eventtypes.NORMAL {
-					commonutils.ManageSpecialEvent(e.Event, commonFlags.Verbose)
-					return ""
-				}
-
-				return parser.TransformEvent(&e)
+				return string(b)
+			case commonutils.OutputModeColumns:
+				fallthrough
+			case commonutils.OutputModeCustomColumns:
+				return parser.TransformIntoColumns(&e)
+			default:
+				fmt.Fprintf(os.Stderr, "Error: %s", commonutils.WrapInErrOutputModeNotSupported(commonFlags.OutputMode))
+				return ""
```

Comment:
Do we really need this check? It's validated [here](https://github.com/inspektor-gadget/inspektor-gadget/blob/46b7d814d7aadd69cf80c65e6f8f5297b89dc823/cmd/common/utils/flags.go#L65). Also, it would be too late to print that error message then there is an event, we should print any error of that kind when the user starts the gadget.

Reply:
Btw, I realized we have this check on many places. Let me remove that.

Reply:
Done in https://github.com/inspektor-gadget/inspektor-gadget/pull/1135.

Reply:
I applied the same approach here.